src/processor/java/org/truffleruby/processor/CoreModuleChecks.java

```diff
@@ -239,4 +242,24 @@ private static boolean isGuarded(String name, String[] guards) {
         }
         return false;
     }
+
+    private static void isParameterBlock(
+            CoreModuleProcessor coreModuleProcessor,
+            VariableElement parameter) {
+        TypeMirror blockType = parameter.asType();
+        Types typeUtils = coreModuleProcessor.getProcessingEnvironment().getTypeUtils();
+        Elements elementUtils = coreModuleProcessor.getProcessingEnvironment().getElementUtils();
+        boolean isNil = typeUtils
+                .isSameType(blockType, elementUtils.getTypeElement("org.truffleruby.language.Nil").asType());
+        boolean isRubyProc = typeUtils
+                .isSameType(blockType, elementUtils.getTypeElement("org.truffleruby.core.proc.RubyProc").asType());
+        boolean isObject = typeUtils.isSameType(blockType, coreModuleProcessor.objectType);
```

Comment:
Looks messy but there isn't much of a better way to do it! It's also offline code.

Reply:
It looks fine, I'll just move the `elementUtils.getTypeElement` in the `CoreModuleProcessor`, unless that doesn't work for some reason?
Maybe also adding `CoreModuleProcessor#isSameType(a, b)` would be nicer, I'll try that.

Reply:
Great, thank you. I was unfamiliar with this API so was definitely fumbling through it — anything tidier you can suggest would be excellent.